testing/test_git.py

```diff
@@ -104,6 +104,70 @@ def test_version_from_git(wd):
     wd("git tag 17.33.0-rc")
     assert wd.version == "17.33.0rc0"
 
+    # custom normalization
+    assert wd.get_version(normalize=False) == "17.33.0-rc"
+    assert wd.get_version(version_cls=NonNormalizedVersion) == "17.33.0-rc"
+    assert (
+        wd.get_version(version_cls="setuptools_scm.NonNormalizedVersion")
+        == "17.33.0-rc"
+    )
+
+
+@pytest.mark.parametrize("with_class", [False, type, str])
+def test_git_version_unnormalized_setuptools(with_class, tmpdir, wd, monkeypatch):
```

Comment:
This is the test that is of interest to me, but it could be moved to the "basic api" test if I have a way to inject the "1.0.0-rc1" version. Indeed there is nothing really specific to git here. See previous comment.

Reply:
it may be possible to jsut use one of the meta tests, we can do this as a followup